include/deal.II/numerics/data_out_dof_data.templates.h

```diff
@@ -1760,27 +1802,25 @@ DataOut_DoFData<DoFHandlerType, patch_dim, patch_space_dim>::
     ranges;
 
   // collect the ranges of dof and cell data
-  using data_iterator = typename std::vector<std::shared_ptr<
-    internal::DataOutImplementation::DataEntryBase<DoFHandlerType>>>::
-    const_iterator;
-
   unsigned int output_component = 0;
-  for (data_iterator d = dof_data.begin(); d != dof_data.end(); ++d)
-    for (unsigned int i = 0; i < (*d)->n_output_variables;)
+  for (const auto &input_data : dof_data)
+    for (unsigned int i = 0; i < input_data->n_output_variables;
+         /* i is updated below */)
       // see what kind of data we have here. note that for the purpose of the
       // current function all we care about is vector data
-      switch ((*d)->data_component_interpretation[i])
+      switch (input_data->data_component_interpretation[i])
         {
           case DataComponentInterpretation::component_is_scalar:
             {
-              // Just move one component forward by one (or two if the component
-              // happens to be complex-valued and we don't use a postprocessor
+              // Just move one component forward by one (or two if the
```

Comment:
```suggestion
              // Just move component forward by one (or two if the
```
? In any case, I was a bit confused.